lib/ssr/category.ts

```diff
@@ -0,0 +1,61 @@
+import { ApolloQueryResult } from "@apollo/client";
+import { ParsedUrlQuery } from "querystring";
+
+import {
+  CategoryPathsDocument,
+  CategoryPathsQuery,
+  CategoryPathsQueryVariables,
+} from "@/saleor/api";
+
+import apolloClient from "../graphql";
+import { CHANNELS, LOCALES, Path } from "../regions";
+
+export interface CategoryPathArguments extends ParsedUrlQuery {
+  channel: string;
+  locale: string;
+  slug: string;
+}
+
+export const categoryPaths = async () => {
+  let paths: Path<CategoryPathArguments>[] = [];
+
+  let hasNextPage = true;
+  let endCursor = "";
+
+  while (hasNextPage) {
+    const response: ApolloQueryResult<CategoryPathsQuery> =
+      await apolloClient.query<CategoryPathsQuery, CategoryPathsQueryVariables>(
+        {
+          query: CategoryPathsDocument,
+          fetchPolicy: "no-cache",
+          variables: {
+            after: endCursor,
+          },
+        }
+      );
+
+    const edges = response.data.categories?.edges;
+    if (!edges) {
+      break;
+    }
+    const responseSlugs: string[] = edges.map((edge) => edge.node.slug);
+    for (let channel of CHANNELS) {
```

Comment:
Why is for loop mixed with forEach? Let's choose a single iterating approach and stick to it:
```
    CHANNELS.forEach((channel) => {
      const channelSlug = channel.slug;

      LOCALES.forEach((locale) => {
        responseSlugs.forEach((slug) => {
          paths.push({
            params: {
              channel: channelSlug,
              locale: locale,
              slug,
            },
          });
        });
      });
    });
```